test/functional/services/data_views.ts

```diff
@@ -0,0 +1,145 @@
+    await this.testSubjects.click('createDataViewButton');
+    await this.create({ name, adHoc, hasTimeField });
+  }
+
+  /**
+   * Returns name for the currently selected Data View
+   */
+  async getSelectedName() {
+    return this.testSubjects.getVisibleText('*dataView-switch-link');
+  }
+
+  /**
+   * Checks if currently selected Data View has temporary badge
+   */
+  async isAdHoc() {
+    const dataView = await this.testSubjects.getAttribute('*dataView-switch-link', 'title');
+    await this.testSubjects.click('*dataView-switch-link');
+    const hasBadge = await this.testSubjects.exists(`dataViewItemTempBadge-${dataView}`);
+    await this.testSubjects.click('*dataView-switch-link');
+    return hasBadge;
+  }
+
+  /**
+   * Opens Create field flayout for the selected Data View
+   */
+  async clickAddFieldFromSearchBar() {
+    await this.testSubjects.click('*dataView-switch-link');
+    await this.testSubjects.click('indexPattern-add-field');
+    await this.testSubjects.existOrFail('fieldEditor');
+  }
+
+  /**
+   * Switch Data View from top search bar
+   */
+  public async switchTo(name: string) {
+    const selectedDataView = await this.getSelectedName();
+    if (name === selectedDataView) {
+      return;
+    }
+    await this.testSubjects.click('*dataView-switch-link');
+    await this.testSubjects.existOrFail('indexPattern-switcher');
+    await this.testSubjects.setValue('indexPattern-switcher--input', name);
+    await this.find.clickByCssSelector(
+      `[data-test-subj="indexPattern-switcher"] [title="${name}"]`
+    );
+  }
+
+  /**
+   * Waits for selected Data View to equal name argument
+   */
+  public async waitForSwitcherToBe(name: string) {
+    await this.retry.waitFor(
+      'Data View switcher to be updated',
+      async () => (await this.getSelectedName()) === name
+    );
+  }
+
+  /**
+   * Switch Data View from top search bar and validate selection is applied
+   */
+  public async switchToAndValidate(name: string) {
+    await this.switchTo(name);
+    await this.waitForSwitcherToBe(name);
+  }
+
+  /**
+   * Edit currently selected Data View
+   */
+  public async editFromSearchBar({
+    newName,
+    newTimeField,
+  }: {
+    newName?: string;
+    newTimeField?: string;
+  }) {
+    await this.testSubjects.click('*dataView-switch-link');
+    await this.testSubjects.click('indexPattern-manage-field');
+    await this.testSubjects.existOrFail('indexPatternEditorFlyout');
+    if (newName) {
+      await this.testSubjects.setValue('createIndexPatternTitleInput', newName, {
+        clearWithKeyboard: true,
+        typeCharByChar: true,
+      });
+    }
+    if (newTimeField) {
+      await this.comboBox.set('timestampField', newTimeField);
+    }
+    await this.testSubjects.click('saveIndexPatternButton');
+    if (await this.testSubjects.exists('confirmModalConfirmButton')) {
+      await this.testSubjects.click('confirmModalConfirmButton');
```

Comment:
@dmlemeshko do we not need assertions here, that assert that the click requested, actually worked?

Reply:
There are other clicks here that make me ask the same question.

Reply:
I copied this logic from `unified_search_page.ts` mostly as is:

```
  public async editDataView(newPattern?: string, newTimeField?: string) {
    await this.clickEditDataView();
    if (newPattern) {
      await this.testSubjects.setValue('createIndexPatternTitleInput', newPattern, {
        clearWithKeyboard: true,
        typeCharByChar: true,
      });
    }
    if (newTimeField) {
      await this.comboBox.set('timestampField', newTimeField);
    }
    await this.testSubjects.click('saveIndexPatternButton');
    if (await this.testSubjects.exists('confirmModalConfirmButton')) {
      await this.testSubjects.click('confirmModalConfirmButton');
    }
  }
```

I wouldn't change as we are not code owners and it might break some existing tests.